### ISSUE TITLE
"Files" sidebar is not expanding folders when navigating to subfolders

### ISSUE THREAD

**Issue by mrnugget** [sourcegraph/sourcegraph#14856]
- **Sourcegraph version:** Sourcegraph.com
- **Platform information:** Chrome

#### Steps to reproduce:

1. Go to https://sourcegraph.com/github.com/sourcegraph/sourcegraph
2. Click on the `cmd` folder in the main view (not the file sidebar!).
3. It looks like this:
<img width="1290" alt="screenshot_2020-10-19_13 42 08@2x" src="https://user-images.githubusercontent.com/1185253/96445886-e7a41900-1210-11eb-9714-054fbe396014.png">
4. Click on a subfolder, like `frontend`.

#### Expected behavior:

I expect `frontend` to be shown in the sidebar and to be also expanded.

#### Actual behavior:

`cmd` is still highlighted, but not expanded and `frontend` is not shown.

<img width="1063" alt="screenshot_2020-10-19_13 43 09@2x" src="https://user-images.githubusercontent.com/1185253/96445972-099d9b80-1211-11eb-86d6-9589dfca0db5.png">

**Comment by attfarhan** [comment]
@felixfbecker @lguychard I built the file tree a couple years back, and can look into fixing this. However, this is a component that should probably be owned by the Web team now instead of Search. @felixfbecker can I assign you to delegate this?

**Comment by felixfbecker** [comment]
Yes, maybe we can look at this when we're working on the directory page. The tree code is very complex though and I personally don't have an understanding of how it works, so would you be open to do a hand-off session with us where you walk us through the code? Or you could join one of our team hack sessions and we can fix it together (and leave with a better understanding and more ownership over the code)

**Comment by attfarhan** [comment]
Sorry I missed this comment @felixfbecker -- yes happy to help in either of those ways.

**Comment by pdubroy** [comment]
Related to #3439

**Comment by jasongornall** [comment]
Closing as I see folders expand now https://sourcegraph.com/github.com/sourcegraph/sourcegraph

**Comment by mrnugget** [comment]
Re-opening because when following the steps in the original ticket folders still don't expand. Am I missing something @jasongornall? 🤔 

https://user-images.githubusercontent.com/1185253/174748562-6f6f4ab4-11ab-4d17-9128-5e6e7bb6740d.mp4

**Comment by sourcegraph-bot-2** [comment]
Heads up @jplahn - the "team/repo-management" label was applied to this issue.

**Comment by mrnugget** [comment]
This is related: we don't expand the file tree when navigating to a file via the fuzzy finder:

https://user-images.githubusercontent.com/1185253/177278392-8b3d680d-8de4-464e-973a-33fb949ba734.mp4

**Comment by macraig** [comment]
@jplahn @jasongornall I think this is either frontend or code intel as it's part of code navigation?

**Comment by mrnugget** [comment]
Pinging @benvenker here, since he now owns core workflow, and I think this is part of the core workflow. I'm personally less interested in finding out who officially owns it, but rather: is it low or high priority (Ben decides) and if it's high priority, who can fix it?
